mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -717,6 +719,53 @@ class SparseVector(
       new SparseVector(size, ii, vv)
     }
   }
+
+  override def argmax: Int = {
+    if (size == 0) {
+      -1
+    } else {
+
+      var maxIdx = indices(0)
+      var maxValue = values(0)
+
+      foreachActive { (i, v) =>
+        if (v > maxValue) {
+          maxIdx = i
+          maxValue = v
+        }
+      }
+
+      // look for inactive values in case all active node values are negative
+      if(size != values.size && maxValue <= 0){
+        val firstInactiveIdx = calcFirstInactiveIdx(0)
+        if(maxValue == 0){
+          if(firstInactiveIdx >= maxIdx) maxIdx else maxIdx = firstInactiveIdx
```

Comment:
I think we can do away with the else here.

if (!(maxValue == 0 && firstInactiveIdx >= maxIdx)) {
    maxIdx = firstInactiveIdx
}